diff -u wvstreams-4.6.1/debian/changelog wvstreams-4.6.1/debian/changelog
--- wvstreams-4.6.1/debian/changelog
+++ wvstreams-4.6.1/debian/changelog
@@ -1,3 +1,13 @@
+wvstreams (4.6.1-1ubuntu1) maverick; urgency=low
+
+  * ipstreams/wvunixdgsocket.cc, streams/wvatomicfile.cc:
+    + remove defenition of MACOS and spurious #endif to fix chmod not 
+      found gcc error and hence build problems (LP: #602134)
+    (forwarded upstream as 
+    http://code.google.com/p/wvstreams/issues/detail?id=28)
+
+ -- Bhavani Shankar <bhavi@ubuntu.com>  Thu, 12 Aug 2010 16:08:57 +0530
+
 wvstreams (4.6.1-1) unstable; urgency=low
 
   * QA upload.
diff -u wvstreams-4.6.1/debian/control wvstreams-4.6.1/debian/control
--- wvstreams-4.6.1/debian/control
+++ wvstreams-4.6.1/debian/control
@@ -1,7 +1,8 @@
 Source: wvstreams
 Section: libs
 Priority: optional
-Maintainer: Debian QA Group <packages@qa.debian.org>
+Maintainer: Ubuntu Core Developers <ubuntu-devel-discuss@lists.ubuntu.com>
+XSBC-Original-Maintainer: Debian QA Group <packages@qa.debian.org>
 Build-Depends: debhelper (>> 5.0.0), quilt, autotools-dev, libdb-dev, libpam0g-dev, libqt3-mt-dev, libssl-dev, libxplc0.3.13-dev, libpopt-dev, zlib1g-dev, doxygen, libdbus-1-dev
 Build-Conflicts: valgrind
 Standards-Version: 3.8.3
only in patch2:
unchanged:
--- wvstreams-4.6.1.orig/ipstreams/wvunixdgsocket.cc
+++ wvstreams-4.6.1/ipstreams/wvunixdgsocket.cc
@@ -1,8 +1,6 @@
 #include "wvunixdgsocket.h"
-#ifdef MACOS
 #include <sys/types.h>
 #include <sys/stat.h>
-#endif
 
 WvUnixDGSocket::WvUnixDGSocket(WvStringParm filename, bool _server, int perms)
     : socketfile(filename)
only in patch2:
unchanged:
--- wvstreams-4.6.1.orig/streams/wvatomicfile.cc
+++ wvstreams-4.6.1/streams/wvatomicfile.cc
@@ -10,10 +10,7 @@
 #include "wvatomicfile.h"
 #include "wvfileutils.h"
 #include "wvstrutils.h"
-
-#ifdef MACOS
 #include <sys/stat.h>
-#endif
 
 WvAtomicFile::WvAtomicFile(WvStringParm filename, int flags, mode_t create_mode)
     : tmp_file(WvString::null)

